web/proxy.js

```diff
@@ -28,30 +25,32 @@ const server = http.createServer((request, response) => {
     let requestPath = request.url;
 
     // Regex not declared globally to avoid internal regex pointer reset for each request.
-    const apiRegex = /\/(staging.*api)/g;
+    const apiRegex = /(\/staging.*)api/g;
 
     /**
      * We only match staging api root to redirect requests to the staging server if the request
-     * is an API call, instead of chat attachments. By default, requests are sent to the prod server.
+     * is an API call, except chat attachments. By default, requests are sent to the prod server.
      * For example,
      * /api?command=OpenReport => request sent to production server
      * /staging-api?command=OpenReport => request sent to staging server
      * /staging-secure-api?command=OpenReport => request sent to secure staging server
      * /chat-attachments/46545... => request sent to production server
      */
-    const apiRootMatch = apiRegex.exec(request.url);
+    const apiRootMatch = apiRegex.exec(requestPath);
 
     // Switch host only if API call, not on chat attachments.
     if (apiRootMatch) {
         const apiRoot = apiRootMatch[1];
-        hostname = HOST_MAP[apiRoot];
+
+        // apiRoot can only be staging or secure staging
+        hostname = apiRoot === proxyConfig.STAGING ? stagingHost : stagingSecureHost;
```

Comment:
I think this is misleading, I think we should invert this so its clear its not Staging/production condition but secure non-secure
```suggestion
        hostname = apiRoot === proxyConfig. STAGING_SECURE ? stagingSecureHost : stagingHost;
```